hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/MergeOnReadIncrementalRelation.scala

```diff
@@ -78,7 +78,16 @@ class MergeOnReadIncrementalRelation(val sqlContext: SQLContext,
   private val tableStructSchema = AvroConversionUtils.convertAvroSchemaToStructType(tableAvroSchema)
   private val maxCompactionMemoryInBytes = getMaxCompactionMemoryInBytes(jobConf)
   private val fileIndex = buildFileIndex()
-
+  private val preCombineField = {
+    val fieldFromTableConfig = metaClient.getTableConfig.getPreCombineField
+    if (fieldFromTableConfig != null) {
+      fieldFromTableConfig
+    } else if (optParams.contains(DataSourceWriteOptions.PRECOMBINE_FIELD_OPT_KEY)) {
```

Comment:
can we use the `HoodieTableConfig` instead? or somehow translate all precombine field options into one place and deprecate others. Using the write option while reading sounds a bit odd.

Reply:
Yes, the PRECOMBINE_FIELD_OPT_KEY is used to compatible with the old table which has not store the preCombineField to hoodie.properties.  Using the write option is a bit odd. So I have add a `READ_PRECOMBINE_FIELD` to the `DataSourceReadOptions`.